AccessCheckoutSDK/AccessCheckoutSDKTests/validation/model/CardValidationStateHandlerTests.swift

```diff
@@ -70,173 +70,222 @@ class CardValidationStateHandlerTests: XCTestCase {
+            cardBrands: [visaBrand, maestroBrand]
+        )
 
-        validationStateHandler.notifyMerchantOfPanValidationState()
-        validationStateHandler.notifyMerchantOfPanValidationState()
+        let brands = validationStateHandler.getCardBrands()
 
-        verify(merchantDelegate, times(1)).panValidChanged(isValid: false)
+        XCTAssertEqual(brands.count, 2)
+        XCTAssertEqual(brands[0].name, "visa")
+        XCTAssertEqual(brands[1].name, "maestro")
     }
 
-    func testGetCardBrand_shouldReturnCardBrandIfKnownBrand() {
+    func testGetCardBrands_shouldReturnEmptyArrayWhenNoBrands() {
         let validationStateHandler = CardValidationStateHandler(
-            merchantDelegate: merchantDelegate, panValidationState: false, cardBrand: visaBrand)
+            merchantDelegate: merchantDelegate,
+            panValidationState: false,
+            cardBrands: []
+        )
+
+        let brands = validationStateHandler.getCardBrands()
 
-        XCTAssertEqual(validationStateHandler.getCardBrand(), visaBrand)
+        XCTAssertTrue(brands.isEmpty)
     }
 
-    func testGetCardBrand_shouldReturnNilCardBrandIfUnknownBrand() {
+    // MARK: PAN Validation State Notification Tests
+
+    func testNotifyMerchantOfPanValidationState_notifiesMerchantOfValidPan() {
         let validationStateHandler = CardValidationStateHandler(
-            merchantDelegate: merchantDelegate, panValidationState: false, cardBrand: nil)
+            merchantDelegate: merchantDelegate, panValidationState: true)
 
-        XCTAssertEqual(validationStateHandler.getCardBrand(), nil)
-    }
+        validationStateHandler.notifyMerchantOfPanValidationState()
 
-    // MARK: ExpiryDateValidationStateHandler
+        verify(merchantDelegate).panValidChanged(isValid: true)
+    }
 
     func
-        testHandleExpiryDateValidation_shouldNotNotifyMerchantDelegateWhenExpiryValidationStateDoesNotChangeFromFalse()
+        testNotifyMerchantOfPanValidationState_notifiesMerchantOfValidPanOnlyOnceWhenCalledMultipleTimes()
     {
         let validationStateHandler = CardValidationStateHandler(
-            merchantDelegate: merchantDelegate,
-            expiryDateValidationState: false
-        )
+            merchantDelegate: merchantDelegate, panValidationState: true)
 
-        validationStateHandler.handleExpiryDateValidation(isValid: false)
-        verify(merchantDelegate, never()).expiryDateValidChanged(isValid: any())
+        validationStateHandler.notifyMerchantOfPanValidationState()
+        validationStateHandler.notifyMerchantOfPanValidationState()
+
+        verify(merchantDelegate, times(1)).panValidChanged(isValid: true)
     }
 
+    // MARK: ExpiryDateValidationStateHandler (unchanged)
+
     func
-        testHandleExpiryDateValidation_shouldNotNotifyMerchantDelegateWhenExpiryValidationStateDoesNotChangeFromTrue()
+        testHandleExpiryDateValidation_shouldNotNotifyMerchantDelegateWhenExpiryValidationStateDoesNotChangeFromFalse()
```

Comment:
review - we are missing
- `testHandleExpiryDateValidation_shouldNotNotifyMerchantDelegateWhenExpiryValidationStateDoesNotChangeFromTrue`
- `testHandleExpiryDateValidation_shouldNotifyMerchantDelegateWhenExpiryValidationStateChangesFromTrue`